### PR TITLE
Add `create` function to kv store

### LINKED ISSUE #1204: Add `create` for async-nats KV
### Proposed change

The go library seems to have the `Create` function which succeeds to add a value for a key if it’s not set, or if the last operation was a delete. 

However, the async-nats rust package seems to missing it.

```go
// Create will add the key/value pair if it does not exist.
Create(key string, value []byte) (revision uint64, err error)
```

### Use case

Leases / Distributed lock

### Contribution

Yes

### PR DESCRIPTION
**Pull request by praveenperera**
Closes #1204

### CODE REVIEW

**Comment by Jarema** [comment]
Thank you for the PR!
I will take a look over it soon.

File: async-nats/src/jetstream/kv/mod.rs [hunk 1]
```diff
@@ -1180,6 +1234,59 @@ impl Display for StatusErrorKind {
...
+}
+
+impl Display for CreateErrorKind {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::AlreadyExists => write!(f, "key already exists"),
+            Self::Publish => write!(f, "failed to put key into store"),
+            Self::Ack => write!(f, "ack error"),
+            Self::InvalidKey => write!(f, "key cannot be empty or start/end with `.`"),
+            Self::Other => write!(f, "other error"),
```
Comment (Jarema) [inline async-nats/src/jetstream/kv/mod.rs:1283]: I would call it "failed creating a key"  to be consistent with other errors.

**Review by Jarema — CHANGES_REQUESTED** [review]
Thank you for your contribution!

Looks good. Just one comment around error Display impl.

File: async-nats/src/jetstream/kv/mod.rs (hunk 1 above)
Comment (praveenperera) [reply async-nats/src/jetstream/kv/mod.rs:1283]: Makes sense, made the change

**Review by Jarema — APPROVED** [review]
LGTM!

**Pull request merged**